packages/redux-dynostore-core/src/dynamicReducers.js

```diff
@@ -6,48 +6,43 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import concatenateReducers from 'redux-concatenate-reducers'
-import filteredReducer from './filteredReducer'
+import mergeReducers from './mergeReducers'
+import filtered from './filteredReducer'
 import createDynamicReducer from './createDynamicReducer'
 import flattenReducers from './flattenReducers'
-import { globalAction } from 'redux-subspace'
-import { DETACH_TYPE } from './detachableReducer'
+import { detachable, detach } from './detachableReducer'
 
 const dynamicReducersEnhancer = () => createHandlers => (store, reducer, ...rest) => {
   let dynamicReducers = {}
 
   const createReducer = () => {
-    const reducers = [
-      filteredReducer(reducer)
-    ]
+    const reducers = [filtered(reducer), createDynamicReducer(dynamicReducers)]
 
-    // Safegaurd against no dynamic reducers
-    if (Object.keys(dynamicReducers).length) {
-      reducers.push(filteredReducer(createDynamicReducer(dynamicReducers)))
-    }
-
-    return concatenateReducers(reducers)
+    return mergeReducers(reducers)
   }
 
   const attachReducers = reducers => {
-    dynamicReducers = { ...dynamicReducers, ...flattenReducers(reducers) }
+    Object.entries(flattenReducers(reducers)).forEach(([identifier, reducer]) => {
+      dynamicReducers[identifier] = detachable(identifier)(reducer)
+    })
     store.replaceReducer(createReducer())
   }
 
-  const detachReducer = identifiers => {
+  const detachReducers = identifiers => {
     identifiers.forEach(identifier => {
```

Comment:
Also obviously happy to apply this during my PR into the main repo to save you the time, if you agree

Reply:
It wouldn't hurt to do it here.  The reason it was required there is that we call a function on the saga itself (`cancel`), so it will error if we don't filter first.  Here, there is no destructive effect from dispatching an unnecessary action into the store, but I'm tempted to say do it for consistency and whatever slight performance improvement might be gained from not doing the wasted work.